includes/Checker/Preparations/Use_Minimal_Theme_Preparation.php

```diff
@@ -0,0 +1,142 @@
+		$this->theme_slug = $theme_slug;
+		$this->themes_dir = $themes_dir;
+	}
+
+	/**
+	 * Runs this preparation step for the environment and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		// Override the theme slug and name.
+		add_filter( 'template', array( $this, 'get_theme_slug' ) );
+		add_filter( 'stylesheet', array( $this, 'get_theme_slug' ) );
+		add_filter( 'pre_option_template', array( $this, 'get_theme_slug' ) );
+		add_filter( 'pre_option_stylesheet', array( $this, 'get_theme_slug' ) );
+		add_filter( 'pre_option_current_theme', array( $this, 'get_theme_name' ) );
+
+		// Override the theme directory.
+		add_filter( 'pre_option_template_root', array( $this, 'get_theme_root' ) );
+		add_filter( 'pre_option_stylesheet_root', array( $this, 'get_theme_root' ) );
+
+		// Register the custom themes directory if relevant.
+		if ( ! empty( $this->themes_dir ) ) {
+			register_theme_directory( $this->themes_dir );
+
+			// Force new directory scan to ensure the new theme directory is available.
+			search_theme_directories( true );
+		}
+
+		// Return the cleanup function.
+		return function() {
+			global $wp_theme_directories;
+
+			remove_filter( 'template', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'stylesheet', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'pre_option_template', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'pre_option_stylesheet', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'pre_option_current_theme', array( $this, 'get_theme_name' ) );
+
+			remove_filter( 'pre_option_template_root', array( $this, 'get_theme_root' ) );
+			remove_filter( 'pre_option_stylesheet_root', array( $this, 'get_theme_root' ) );
+
+			if ( ! empty( $this->themes_dir ) ) {
+				$index = array_search( untrailingslashit( $this->themes_dir ), $wp_theme_directories, true );
+				if ( false !== $index ) {
+					array_splice( $wp_theme_directories, $index, 1 );
+					$wp_theme_directories = array_values( $wp_theme_directories );
+				}
+			}
+		};
+	}
+
+	/**
+	 * Gets the theme slug.
+	 *
+	 * Used as a filter callback.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The theme slug.
+	 */
+	public function get_theme_slug() {
+		return $this->theme_slug;
+	}
+
+	/**
+	 * Gets the theme name.
+	 *
+	 * Used as a filter callback.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The theme name.
+	 */
+	public function get_theme_name() {
+		$theme = wp_get_theme( $this->theme_slug, $this->themes_dir );
+		return $theme->display( 'Name' );
+	}
+
+	/**
+	 * Gets the theme root.
+	 *
+	 * Used as a filter callback.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The theme root.
+	 */
+	public function get_theme_root() {
+		return get_raw_theme_root( $this->theme_slug, true );
+	}
+}
```

Comment:
Can you add unit tests for this class?

Reply:
@felixarntz unit tests are setup here https://github.com/10up/plugin-check/pull/56/files#diff-f97377ea014b79e48c71fe98cb7b27e474c74960dd44538516850361badd7ac9

Might have been difficult to see with all the theme files included in the changes.

Reply:
@jjgrainger Apologies, missed that indeed!